### PR TITLE
Update ZIO to RC19

### PR DESCRIPTION
**Pull request by ghostdogpr**

### CODE REVIEW

**Comment by iravid** [comment]
The hanging test is `acquisition of Reservation preserves cancellability in new F` in `CatsZManagedSyntaxSpec`. Some observations:
- The acquisition IO in that test is executed interruptibly (verified with `ZIO.checkInterruptible`).
- The test is hanging on the call to `f.cancel`.

This is pretty tricky to diagnose because the fiber is a cats IO fiber which is running ZIO code lifted by `CatsConcurrentEffect#toIO`.

Bowing out for now.

**Comment by neko-kai** [comment]
@ghostdogpr @iravid 
I think I fixed `acquisition of Reservation preserves cancellability in new F`.
I also added forced uninterruptibility for acquire/release in ZManaged.bracketCase.

File: interop-cats/shared/src/main/scala/zio/interop/catszmanaged.scala [hunk 1]
```diff
@@ -65,28 +71,37 @@ final class CatsIOResourceSyntax[F[_], A](private val resource: Resource[F, A])
...
 }
 
 final class ZManagedSyntax[R, E, A](private val managed: ZManaged[R, E, A]) extends AnyVal {
 
-  def toResourceZIO(implicit ev: Applicative[ZIO[R, E, ?]]): Resource[ZIO[R, E, ?], A] =
-    Resource
-      .makeCase(managed.reserve)((r, e) => r.release(exitCaseToExit(e)).unit)
-      .evalMap(_.acquire)
+  def toResourceZIO: Resource[ZIO[R, E, ?], A] =
+    Resource.suspend {
```
Comment (iravid) [inline interop-cats/shared/src/main/scala/zio/interop/catszmanaged.scala:85]: Ahh nice. `Resource.Allocate` is always uninterruptible. Makes sense!

**Comment by iravid** [comment]
Thanks for figuring that one out @neko-kai!

**Pull request merged**